### PR TITLE
[ci] add clang-tidy

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

File: .github/workflows/clang-tidy.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+on:
+  push:
+    branches: [ master, stable*  ]
+  pull_request:
+    branches: [ master, stable* ]
+
+permissions:
+  contents: read
+  issues: write
+  pull-requests: write
```
Comment (bwrsandman) [inline .github/workflows/clang-tidy.yml:11]: Normally not needed

File: .github/workflows/clang-tidy.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: clang-tidy-review
+on:
+  push:
+    branches: [ master, stable*  ]
```
Comment (bwrsandman) [inline .github/workflows/clang-tidy.yml:4]: clang-tidy review is meant to run on PRs not pushes.

**Review by bwrsandman — CHANGES_REQUESTED** [review]
The fact that it's not positng might be that in your PR's context, it doesn't have access to the token from FreeRDP since it's running in akallabeth's context.
Not completely sure though.

When [I run it](https://github.com/openblack/openblack/blob/master/.github/workflows/ci-vcpkg.yml#L172C1-L178C29), it's in the [`pull_request_target`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target) trigger which is a bit different from `pull_request` in that the tests come from the target and not the branch of the PR.

File: .github/workflows/clang-tidy.yml (hunk 2 above)
Comment (akallabeth) [reply .github/workflows/clang-tidy.yml]: ok, removed.

File: .github/workflows/clang-tidy.yml (hunk 1 above)
Comment (akallabeth) [reply .github/workflows/clang-tidy.yml]: ok, removed.

**Comment by akallabeth** [comment]
@freerdp-bot test please

**Pull request merged**